common/persistence/cassandraPersistence.go

```diff
@@ -2272,7 +2278,7 @@ func (d *cassandraPersistence) GetTasks(request *GetTasksRequest) (*GetTasksResp
 		rowTypeTask,
 		request.ReadLevel,
 		request.MaxReadLevel,
-		request.BatchSize)
+	).PageSize(request.BatchSize)
```

Comment:
You are using paginated query but not returning pageToken back to the caller.

Reply:
this is for the task, by the matching side.
for anything other than timer, we use the task id for querying